### PR TITLE
Issue #195 : When uploading, warn when the user quits the page

### PR DESCRIPTION
**Pull request by DimitriGilbert**

### CODE REVIEW

File: client/src/app/videos/+video-edit/video-upload-guard.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+@Injectable()
+export class VideoUploadGuard implements CanDeactivate<VideoAddComponent> {
+  canDeactivate (component: VideoAddComponent,
+    currentRoute: ActivatedRouteSnapshot,
+    currentState: RouterStateSnapshot,
+    nextState: RouterStateSnapshot
+  ): boolean {
+    console.log('VideoUploadGuard')
+    console.log(currentRoute.params)
+    console.log(currentState.url)
```
Comment (bnjbvr) [inline client/src/app/videos/+video-edit/video-upload-guard.ts]: trivia: debug spew to remove?

File: client/src/app/videos/+video-edit/video-upload-guard.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    return component.canDeactivate() || window.confirm('Your upload will be canceled, are you sure?')
+    // return true;
```
Comment (bnjbvr) [inline client/src/app/videos/+video-edit/video-upload-guard.ts]: trivia: remove this comment

**Comment by DimitriGilbert** [comment]
very true, I forgot my debug stuff

File: client/src/app/videos/+video-edit/video-upload-guard.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { VideoAddComponent } from './video-add.component'
+
+@Injectable()
+export class VideoUploadGuard implements CanDeactivate<VideoAddComponent> {
+  canDeactivate (component: VideoAddComponent,
+    currentRoute: ActivatedRouteSnapshot,
+    currentState: RouterStateSnapshot,
+    nextState: RouterStateSnapshot
+  ): boolean {
+    return component.canDeactivate() || window.confirm('Your upload will be canceled, are you sure?')
```
Comment (Chocobozzz) [inline client/src/app/videos/+video-edit/video-upload-guard.ts]: Maybe you could use the `confirmService`? :)

Usage example: https://github.com/Chocobozzz/PeerTube/blob/develop/client/src/app/videos/%2Bvideo-watch/video-watch.component.ts#L127

**Comment by DimitriGilbert** [comment]
Fairly new to angular, I didn't know there was a confirm service ^^, Keep learning every day :)

**Comment by Chocobozzz** [comment]
I think it would be better to create an universal `canDeactivateGuard` like in this article: https://scotch.io/courses/routing-angular-2-applications/candeactivate

Then you would just move the confirmation dialog in the watch service :)

**Comment by Chocobozzz** [comment]
Thanks @DimitriGilbert!

**Pull request merged**